SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -409,365 +231,163 @@ void FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
 		return;
 	}
 
-	LocalRunningDeploymentID.Empty();
-
 	bStartingDeployment = true;
 
-	// Stop the currently running service if the runtime IP is to be exposed, but is different from the one specified
-	if (ExposedRuntimeIP != RuntimeIPToExpose)
-	{
-		UE_LOG(LogSpatialDeploymentManager, Verbose,
-			   TEXT("Settings for exposing runtime IP have changed since service startup. Restarting service to reflect changes."));
-		TryStopSpatialService();
-	}
-
-	// If the service is not running then start it.
-	if (!bSpatialServiceRunning)
-	{
-		TryStartSpatialService(RuntimeIPToExpose);
-	}
+	FString SchemaBundle = SpatialGDKServicesConstants::SchemaBundlePath;
 
-	SnapshotName.RemoveFromEnd(TEXT(".snapshot"));
+	// Give the snapshot path a timestamp to ensure we don't overwrite snapshots from older deployments.
+	// The snapshot service saves snapshots with the name `snapshot-n.snapshot` for a given deployment,
+	// where 'n' is the number of snapshots taken since starting the deployment.
+	FString SnapshotPath = FPaths::Combine(SpatialGDKServicesConstants::SpatialOSSnapshotFolderPath, *RuntimeStartTime.ToString());
 
-	AttemptSpatialAuthResult = Async(
-		EAsyncExecution::Thread,
-		[this]() {
-			return SpatialCommandUtils::AttemptSpatialAuth(bIsInChina);
-		},
-		[this, LaunchConfig, RuntimeVersion, LaunchArgs, SnapshotName, RuntimeIPToExpose, CallBack]() {
-			bool bSuccess = AttemptSpatialAuthResult.IsReady() && AttemptSpatialAuthResult.Get() == true;
-			if (bSuccess)
-			{
-				bSuccess = FinishLocalDeployment(LaunchConfig, RuntimeVersion, LaunchArgs, SnapshotName, RuntimeIPToExpose);
-			}
-			else
-			{
-				UE_LOG(LogSpatialDeploymentManager, Error,
-					   TEXT("Failed to authenticate against SpatialOS while attempting to start a local deployment."));
-			}
-			bStartingDeployment = false;
+	// Create the folder for storing the snapshots.
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	PlatformFile.CreateDirectoryTree(*SnapshotPath);
 
-			if (CallBack)
-			{
-				CallBack(bSuccess);
-			}
-		});
+	// Use the runtime start timestamp as the log directory, e.g. `<Project>/spatial/localdeployment/<timestamp>/`
+	FString LocalDeploymentLogsDir = FPaths::Combine(SpatialGDKServicesConstants::LocalDeploymentLogsDir, RuntimeStartTime.ToString());
```

Comment:
how is the DateTime -> String conversion done? I am being haunted by the batch issues we had where the user's locale impacted how times were written and some of them cannot be directory names (e.g. 01/12/2020). But maybe Unreal always uses a sane format.

Reply:
It's always of the default format `2020.12.02-12.15.09`. Looks good